instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceResolver.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.net;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PeerServiceResolver {
+
+  private static final Map<String, Set<ServiceMatcher>> MAPPING = new ConcurrentHashMap<>();
+
+  public PeerServiceResolver(Map<String, String> peerServiceMapping) {
+    peerServiceMapping.forEach(
+        (key, serviceName) -> {
+          try {
+            String protocol = null;
+            URI uri;
+            if (key.contains("://")) {
+              uri = new URI(key);
+              protocol = uri.getScheme();
+            } else {
+              uri = new URI("http://" + key);
+            }
+            String host = uri.getHost();
+            Integer port = uri.getPort() != -1 ? Integer.valueOf(uri.getPort()) : null;
+            Set<ServiceMatcher> matchers = MAPPING.get(host);
+            ServiceMatcher serviceMatcher =
+                new ServiceMatcher(serviceName, protocol, port, uri.getPath());
+            if (matchers == null) {
+              matchers = new HashSet<>();
+              matchers.add(serviceMatcher);
+              MAPPING.put(host, matchers);
+            } else {
+              matchers.add(serviceMatcher);
+            }
+          } catch (URISyntaxException use) {
+            // TODO wire logging
+          }
+        });
+  }
+
+  public boolean isEmpty() {
+    return MAPPING.isEmpty();
+  }
+
+  public String resolveService(String protocol, String host, Integer port, String path) {
+    Set<ServiceMatcher> matchers = MAPPING.get(host);
+    if (matchers == null) {
+      return null;
+    } else {
+      return matchers.stream()
+          .filter(serviceMatcher -> serviceMatcher.matches(protocol, port, path))
+          .max(Comparator.comparingInt(ServiceMatcher::getPathLength))
+          .map(ServiceMatcher::getServiceName)
+          .orElse(null);
+    }
+  }
+
+  private static class ServiceMatcher {
+
+    private final String serviceName;
+    private final String protocol;
+    private final Integer port;
+    private final String path;
+
+    public ServiceMatcher(String serviceName, String protocol, Integer port, String path) {
+      this.serviceName = serviceName;
+      this.protocol = protocol;
+      this.port = port;
+      this.path = path;
+    }
+
+    public String getServiceName() {
```

Comment:
Looks like this isn't used. Let's remove it and the `serviceName` field as well.

Reply:
This is used on line 63.